### PR TITLE
DOC: adding a section on NiftiSpheresMasker

### PR DESCRIPTION
**Pull request by mrahim**
Fix issue #863

### CODE REVIEW

File: doc/manipulating_visualizing/data_preparation.rst [hunk 1]
```diff
@@ -436,3 +435,32 @@ possible option.
...
+seeds containing voxels in spheres. It makes it easy to get these signals once
+you have a list of coordinates.
+
+Seeds definition
+------------------
+
+A single seed is a sphere defined by the radius (in millimeters) and the
+coordinates (typically MNI or TAL) of its center.
+
+A set of seeds is thus defined by a list of 3D coordinates and a radius.
```
Comment (GaelVaroquaux) [inline doc/manipulating_visualizing/data_preparation.rst]: Can you give an example here: one line of code that shows the typical input.

The reason being that for a user beginner in Python, it may not be clear what you are talking about.

File: doc/manipulating_visualizing/data_preparation.rst [hunk 2]
```diff
@@ -436,3 +435,32 @@ possible option.
...
+Seeds can overlap spatially and are represented in a binary present-nonpresent
+coding (no weighting).
+
+:class:`NiftiSpheresMasker` Usage
+---------------------------------
+
+"seeds" argument takes a list of 3D coordinates of the spheres centers,
+they should be in the same space as the images (typically MNI or TAL).
```
Comment (GaelVaroquaux) [inline doc/manipulating_visualizing/data_preparation.rst]: It seems that this is redundant with the section before. I wonder if we should have 3 subsections, or just two paragraphs.

File: doc/manipulating_visualizing/data_preparation.rst [hunk 3]
```diff
@@ -436,3 +435,32 @@ possible option.
...
+"radius" is an optional argument that takes a real value in millimeters.
+If no value is given, the signals extracted are from voxels where
+"seeds" have been defined.
```
Comment (GaelVaroquaux) [inline doc/manipulating_visualizing/data_preparation.rst]: I am sorry, I don't understand the last sentence.

File: doc/manipulating_visualizing/data_preparation.rst (hunk 2 above)
Comment (mrahim) [inline doc/manipulating_visualizing/data_preparation.rst]: I agree. I tried to follow the previous sections but two paragraphs might be enough.

Comment (GaelVaroquaux) [inline doc/manipulating_visualizing/data_preparation.rst]: > I tried to follow the previous sections but two paragraphs might be enough.

Don't feel obliged.

File: doc/manipulating_visualizing/data_preparation.rst [hunk 4]
```diff
@@ -436,3 +435,29 @@ possible option.
...
+Using :class:`NiftiSpheresMasker` needs to define a list of coordinates.
+"seeds" argument takes a list of 3D coordinates (tuples) of the spheres centers,
+they should be in the same space as the images.
+Seeds can overlap spatially and are represented in a binary present-nonpresent
+coding (no weighting).
+Below is an example of a coordinates list of four seeds from the default mode network.
+
+.. literalinclude:: ../../examples/connectivity/plot_adhd_spheres.py
+    :start-after: # Coordinates of Default Mode Network
+    :end-before: labels = [
```
Comment (GaelVaroquaux) [inline doc/manipulating_visualizing/data_preparation.rst]: I was thinking to give an actual example here, maybe copy-pasting the code from the example.

The reason being that the 'start-after' / 'end-before' tend to break as soon as we do a refactoring of the examples, and we don't see it.

File: doc/manipulating_visualizing/data_preparation.rst [hunk 5]
```diff
@@ -436,3 +435,29 @@ possible option.
...
+
+"radius" is an optional argument that takes a real value in millimeters.
+If no value is given the spheres are considered as voxels.
```
Comment (GaelVaroquaux) [inline doc/manipulating_visualizing/data_preparation.rst]: I finally understood what you meant. I think that I would rephrase this as:

If no value is given for the radius argument, the single voxel at the given locus is used.

Comment (GaelVaroquaux) [inline doc/manipulating_visualizing/data_preparation.rst]: s/locus/seed position/

File: doc/manipulating_visualizing/data_preparation.rst [hunk 6]
```diff
@@ -436,3 +435,28 @@ possible option.
...
+coordinates (typically MNI or TAL) of its center.
+
+Using :class:`NiftiSpheresMasker` needs to define a list of coordinates.
+"seeds" argument takes a list of 3D coordinates (tuples) of the spheres centers,
+they should be in the same space as the images.
+Seeds can overlap spatially and are represented in a binary present-nonpresent
+coding (no weighting).
+Below is an example of a coordinates list of four seeds from the default mode network.::
+
+  dmn_coords = [(0, -52, 18), (-46, -68, 32), (46, -68, 32), (0, 50, -5)]
```
Comment (GaelVaroquaux) [inline doc/manipulating_visualizing/data_preparation.rst]: Can you add '>>>' at the beginning of the line, that way it get run by doctests.

**Comment by GaelVaroquaux** [comment]
Thanks. Merging.
